definitions/procedures/puppet/enable_puppet.rb

```diff
@@ -0,0 +1,33 @@
+module Procedures::Puppet
+  class EnablePuppet < ForemanMaintain::Procedure
+    metadata do
+      description 'Enable Puppet feature'
+    end
+
+    def run
+      feature(:installer).run(installer_arguments_enabling_puppet.join(' '), :interactive => false)
+    end
+
+    private
+
+    # TODO: This works only for a katello scenario, we should consider vanilla foreman
+    def installer_arguments_enabling_puppet
+      answers = feature(:installer).answers
+
+      options = ['--enable-puppet', '--puppet-server true']
+      if answers.key?('foreman_proxy')
+        options << '--foreman-proxy-puppet true'
+        options << '--foreman-proxy-puppetca true'
+      end
+      if answers.key?('foreman_proxy_content')
+        options << '--foreman-proxy-content-puppet true'
+        options << '--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt'
+        options << '--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt'
+        options << '--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key'
+      end
```

Comment:
I think it's safe to assume this key exists (it should in every scenario). The question is what we do if it's disabled. In that case, enabling these two options don't make sense. If you don't enable these two options, are not passed in, passing in `--enable-puppet` (and friends) don't make sense either. So it becomes:

```suggestion
      options = []
      
      if answers['foreman_proxy']
        options += [
          '--enable-puppet',
          '--puppet-server true',
          '--foreman-proxy-puppet true',
          '--foreman-proxy-puppetca true'
        ]
        if answers.key?('foreman_proxy_content')
          options << '--foreman-proxy-content-puppet true'
          options << '--puppet-server-foreman-ssl-ca /etc/pki/katello/puppet/puppet_client_ca.crt'
          options << '--puppet-server-foreman-ssl-cert /etc/pki/katello/puppet/puppet_client.crt'
          options << '--puppet-server-foreman-ssl-key /etc/pki/katello/puppet/puppet_client.key'
        end
      end
```

One other thing I'm wondering about is how to deal with a Puppet setup without a CA. This can't deal with compile only servers since we're explicitly enabling the Foreman Proxy PuppetCA plugin. Should we also pass `--puppet-server-ca true`? Can a scenario support parameters like `--with-ca` or `--without-ca`?

Reply:
Marking this resolved as enable is out of scope.